sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/PrivateService.java

```diff
@@ -119,15 +76,14 @@ public static <T> Optional<T> extractCompasPrivate(TPrivate tPrivate, Class<T> c
      * @throws ScdException throws when inconsistency between types, or when more than 1 compas element is found
      */
     public static <T> Optional<T> extractCompasPrivate(TBaseElement baseElement, Class<T> compasClass) throws ScdException {
-        List<T> compasPrivates = extractCompasPrivates(baseElement, compasClass);
-        if (compasPrivates.size() > 1) {
-            throw new ScdException(String.format("Expecting maximum 1 private of type %s with 1 element, but found %d",
-                PrivateEnum.fromClass(compasClass).getPrivateType(), compasPrivates.size()));
-        }
-        if (compasPrivates.isEmpty()) {
-            return Optional.empty();
-        }
-        return Optional.of(compasPrivates.get(0));
+        return extractCompasPrivates(baseElement, compasClass)
+                .reduce((t, t2) -> {
+                    //Vérification du nombre d'élements de même type dans le contenu du private => est-ce vraiment necessaire ??
+                    if (t.getClass() == t2.getClass()) {
```

Comment:
Why checking class equality ? `t` et `t2` are if class `T` specified by compasClass (or at least inherit its). There is no inheritance between TCompasXXX classes, so it is safe to remove the test.